src/server/serve_dir.rs

```diff
@@ -31,7 +31,7 @@ impl<State> Endpoint<State> for ServeDir {
         log::info!("Requested file: {:?}", dir);
 
         Box::pin(async move {
-            let file = match async_std::fs::canonicalize(&dir).await {
+            let (file, path) = match async_std::fs::canonicalize(&dir).await {
```

Comment:
Sending back a tuple here feels a bit like a hack. I think inlining the mime guessing, while messy, at least won't change the overall code flow. We could even extract it to a function like:

```rust
fn guess_mime(headers: impl AsMut<Headers>, path: &Path) {
    // guess mime here
}
```
That we then call inline.

Reply:
What if instead of `(file, path)` this returned a named `FileData` struct as mentioned in https://github.com/http-rs/tide/issues/449?